lib/charms/ovn_charm.py

```diff
@@ -1432,3 +1434,19 @@ def assess_exporter(self):
         else:
             snap.install('prometheus-ovs-exporter', channel=channel)
         snap.connect_all()
+
+    def restart_exporter(self, only_inactive=False):
+        """Restart OVS prometheus exporter service.
+
+        :param only_inactive: If set to True, and the service is already
+                              active, it wont' be restarted.
+        :type only_inactive: bool
+        :returns: None
+        """
+        restart_action = (ch_core.host.service_start
+                          if only_inactive
```

Comment:
So while the outcome of issuing `systemctl start` for an already running service is effectively a NO-OP, it does make the systemd process do a lot of unnecessary work, like iterating through all its service configuration and so on.

Whereas a `sytemctl is-active`, as represented by `ch_core.host.service_running`, call is very light weight in comparison. (You can enable debug logging for `systemd` with `systemd-analyze set-log-level debug` and see for yourself.)

This approach also makes the logging below (see next comment) misleading.  Should we perhaps gate this on whether the service is running?

Reply:
hmm, that is an interesting point. Indeed I thought that it was pretty much no-op.